kyo-core/js/src/main/scala/kyo/KyoAppBasePlatformSpecific.scala

```diff
@@ -0,0 +1,42 @@
+package kyo
+
+abstract class KyoAppBasePlatformSpecific extends KyoApp.Base[Throwable, Async & Resource]:
+    override def timeout: Duration = 10.minutes
+
+    protected def handle[A: Flat](v: A < (Async & Resource & Abort[Throwable]))(using Frame): A < (Async & Abort[Throwable]) =
+        Resource.run(v)
+
+    final override protected def args: Array[String] = _args
+
+    private var _args: Array[String] = null
+
+    private var concatenatedAsync: Any < (Async & Abort[Throwable]) = scala.compiletime.uninitialized
+    private var initCode: () => Unit                                = scala.compiletime.uninitialized
+
+    private def keepAlive(using Frame): Unit < (Async & Abort[Throwable]) = Async.sleep(timeout).map(_ => keepAlive)
+
+    final def main(args: Array[String]) =
+        this._args = args
+        if initCode ne null then initCode()
+
+    override protected def run[A: Flat](v: => A < (Async & Resource & Abort[Throwable]))(using Frame): Unit =
+        import AllowUnsafe.embrace.danger
+
+        if null eq concatenatedAsync.asInstanceOf[Object] then
+            concatenatedAsync = handle(v)
+        else
+            concatenatedAsync = concatenatedAsync.map(_ => handle(v))
+        end if
+
+        initCode = (
+            () =>
+                if null ne concatenatedAsync.asInstanceOf[Object] then
+                    val finalAsync = Async.race(concatenatedAsync.map(_ => ()), keepAlive)
+                    val _          = IO.Unsafe.run(Async.run(finalAsync)).eval
+                else
+                    (
+                )
```

Comment:
In JVM we have blocking, so initCode can be sequentially run. In JS we can’t block, so if we run them like we did in JVM, they’ll be spawned as JS promise by the runtime, hence the order of execution is not guaranteed. That why in JS code I concatenate the initCode into a single one.
I didn’t do this concatenation in main method because it requires Frames implicitly, and I was not sure if I want to use Frames.internal or not